src/modules/70-pipeline/components/PipelineStudio/SavePipelinePopover/SavePipelinePopover.tsx

```diff
@@ -0,0 +1,453 @@
+import { validateJSONWithSchema } from '@common/utils/YamlUtils'
+import { useQueryParams } from '@common/hooks'
+import type {
+  GitQueryParams,
+  PathFn,
+  PipelinePathProps,
+  PipelineStudioQueryParams,
+  PipelineType
+} from '@common/interfaces/RouteInterfaces'
+import { useTelemetry } from '@common/hooks/useTelemetry'
+import { usePipelineSchema } from '@pipeline/components/PipelineStudio/PipelineSchema/PipelineSchemaContext'
+import { useSaveAsTemplate } from '@pipeline/components/PipelineStudio/SaveTemplateButton/useSaveAsTemplate'
+import { AppStoreContext } from 'framework/AppStore/AppStoreContext'
+import type { PipelineInfoConfig } from 'services/cd-ng'
+import { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
+import type { GovernanceMetadata } from 'services/pipeline-ng'
+import PipelineErrors from '@pipeline/components/PipelineStudio/PipelineCanvas/PipelineErrors/PipelineErrors'
+import type { AccessControlCheckError } from 'services/rbac'
+import useRBACError from '@rbac/utils/useRBACError/useRBACError'
+import css from './SavePipelinePopover.module.scss'
+
+export interface SavePipelinePopoverProps extends PopoverProps {
+  className?: string
+  toPipelineStudio: PathFn<PipelineType<PipelinePathProps> & PipelineStudioQueryParams>
+  setGovernanceMetadata: (data: GovernanceMetadata | undefined) => void
+  isValidYaml: () => boolean
+}
+
+interface SavePipelineObj {
+  pipeline: PipelineInfoConfig
+}
+
+interface SavePipelineMenuItem {
+  icon?: IconName
+  label: string
+  onClick: () => void
+  disabled?: boolean
+}
+
+export function SavePipelinePopover({
+  toPipelineStudio,
+  setGovernanceMetadata,
+  isValidYaml,
+  className = '',
+  portalClassName,
+  ...popoverProps
+}: SavePipelinePopoverProps): React.ReactElement {
+  const { isGitSyncEnabled } = React.useContext(AppStoreContext)
+  const {
+    state: { pipeline, yamlHandler, gitDetails, pipelineView, isUpdated },
+    deletePipelineCache,
+    fetchPipeline,
+    view,
+    setSchemaErrorView,
+    setSelection,
+    isReadonly,
+    updatePipelineView
+  } = usePipelineContext()
+  const [loading, setLoading] = React.useState<boolean>()
+  const { branch } = useQueryParams<GitQueryParams>()
+  const { trackEvent } = useTelemetry()
+  const { pipelineSchema } = usePipelineSchema()
+  const { showSuccess, showError, clear } = useToaster()
+  const { getRBACErrorMessage } = useRBACError()
+  const { getString } = useStrings()
+  const { OPA_PIPELINE_GOVERNANCE } = useFeatureFlags()
+  const history = useHistory()
+  const { projectIdentifier, orgIdentifier, accountId, pipelineIdentifier, module } =
+    useParams<PipelineType<PipelinePathProps>>()
+  const isYaml = view === SelectedView.YAML
+  const [menuOpen, setMenuOpen] = React.useState(false)
+  const templatesFeatureFlagEnabled = useFeatureFlag(FeatureFlag.NG_TEMPLATES)
+  const [updatePipelineAPIResponse, setUpdatePipelineAPIResponse] = React.useState<any>()
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232314)
```suggestion
  const [updatePipelineAPIResponse, setUpdatePipelineAPIResponse] = React.useState<unknown>()
```